src/App.js

```diff
@@ -21,94 +31,69 @@ const Container = styled.div`
-
-  /**
-   * Store all messages in the conversation
-   * @property {number} id - ID of the user who sent the message.
-   * @property {string} msg - Contents of the message.
-   */
-  const [messages, setMessages] = useState([]);
-
-  /**
-   * Change active user when user clicks on header bar
-   */
-  const handleActiveUserClick = useCallback(() => {
-    if (activeUser.id === 1) {
-      setActiveUser(participants[0]);
-    } else {
-      setActiveUser(participants[1]);
-    }
-  },[activeUser, setActiveUser])
-
-  /**
-   * Adds a new message to the conversation
-   * @param {int} userId - The ID of the user who sent the message
-   * @param {string} msg - The contents of the message
-   */
-  const addNewMessage = useCallback((userId, msg) => {
-    setMessages(prevMessages => [...prevMessages, { id: userId, msg }]);
-  }, [setMessages])
-
-  /**
-   * Handles send action of message. Receives states lifted from ChatForm
-   * @param {string} msg - Message that the user entered into ChatForm > ChatInput
-   */
-  const handleChatSend = useCallback((msg) => {
-    addNewMessage(activeUser.id, msg);
-  }, [activeUser.id, addNewMessage])
-
   return (
-    <Wrapper>
-      <Container>
-        <ActiveUser 
-          handleActiveUserClick={ handleActiveUserClick }
-          activeUser={ activeUser } 
-        />
-        <ChatContainer 
-          participants={ participants } 
-          messages={ messages }
-        />
-        <ChatForm 
-          handleChatSend={ handleChatSend }
-          activeUser={ activeUser } 
-        />
-      </Container>
-    </Wrapper>
+    <ContactsProvider>
+      <RoomsProvider>
+        <Wrapper>
+          <Container>
+            <Navbar />
+            <Content>
+              <Switch>
+                <Route path="/room/:id">
+                  <RequireLogin>
+                    <ChatRoom />
+                  </RequireLogin>
+                </Route>
+                <Route path="/rooms">
+                  <RequireLogin>
+                    <ChatRoomList />
+                  </RequireLogin>
+                </Route>
+                <Route path="/friend/:id">
+                  <FriendProfile />
+                </Route>
+                <Route path="/friends">
+                  <RequireLogin>
+                    <FriendList />
+                  </RequireLogin>
+                </Route>
+                <Route path="/settings">
+                  <RequireLogin>
+                    <Settings />
+                  </RequireLogin>
+                </Route>
+                <Route path="/signup">
+                  <Signup />
+                </Route>
+                <Route path="/auth/login">
+                  <Login />
+                </Route>
+                <Route exact path="/">
+                  <Home />
+                </Route>
+                <Route path="*">
+                  <Page404 />
+                </Route>
```

Comment:
```suggestion
                <Route path="/">
                  <Page404 />
                </Route>
```

요렇게 해도 괜찮을것같아요~!
그리고 `<NotFoundPage>`로 바꿔도 괜찮을듯?